Realm/ObjectServerTests/SwiftFlexibleSyncServerTests.swift

```diff
@@ -52,999 +57,468 @@ class SwiftFlexibleSyncTests: SwiftSyncTestCase {
+            }
         }
-        XCTAssertEqual(subscriptions.count, 2)
-
-        let foundSubscription1 = subscriptions.first(named: "person_age_15")
-        XCTAssertNotNil(foundSubscription1)
-        XCTAssertEqual(foundSubscription1!.name, "person_age_15")
 
-        let foundSubscription2 = subscriptions.first(named: "person_age_20")
-        XCTAssertNotNil(foundSubscription2)
-        XCTAssertEqual(foundSubscription2!.name, "person_age_20")
-    }
-
-    func testFindSubscriptionByQuery() throws {
-        let realm = try openFlexibleSyncRealm()
-        let subscriptions = realm.subscriptions
-        subscriptions.write {
-            subscriptions.append(QuerySubscription<SwiftPerson>(name: "person_firstname_james") {
-                $0.firstName == "James"
-            })
-            subscriptions.append(QuerySubscription<SwiftTypesSyncObject>(name: "object_int_more_than_zero") {
-                $0.intCol > 0
-            })
-        }
-        XCTAssertEqual(subscriptions.count, 2)
-
-        let foundSubscription1 = subscriptions.first(ofType: SwiftPerson.self, where: {
-            $0.firstName == "James"
-        })
-        XCTAssertNotNil(foundSubscription1)
-        XCTAssertEqual(foundSubscription1!.name, "person_firstname_james")
-
-        let foundSubscription2 = subscriptions.first(ofType: SwiftTypesSyncObject.self, where: {
-            $0.intCol > 0
-        })
-        XCTAssertNotNil(foundSubscription2)
-        XCTAssertEqual(foundSubscription2!.name, "object_int_more_than_zero")
-    }
+        let user = try await self.flexibleSyncApp.login(credentials: basicCredentials(usernameSuffix: "", app: flexibleSyncApp))
+        let realm = try user.realm(configuration: Realm.Configuration(objectTypes: [SwiftPerson.self, SwiftDog.self, Bird.self]))
+        let persons: QueryResults<SwiftPerson> = try await realm.subscriptions.subscribe(to: { $0.age > 18 && $0.firstName == "\(#function)" })
+        waitForDownloads(for: realm)
+        XCTAssertEqual(persons.count, 3)
 
-    func testRemoveSubscriptionByName() throws {
-        let realm = try openFlexibleSyncRealm()
-        let subscriptions = realm.subscriptions
-        subscriptions.write {
-            subscriptions.append(QuerySubscription<SwiftPerson>(name: "person_firstname_james") {
-                $0.firstName == "James"
-            })
-            subscriptions.append(
-                QuerySubscription<SwiftTypesSyncObject>(name: "object_int_more_than_zero") {
-                    $0.intCol > 0
-                },
-                QuerySubscription<SwiftTypesSyncObject>(name: "object_string") {
-                    $0.stringCol == "John" || $0.stringCol == "Tom"
-                })
-        }
-        XCTAssertEqual(subscriptions.count, 3)
+        // This will trigger a client reset, which will result in the server not responding to any instruction, this is not removing the object from the database.
+//        let newPerson = SwiftPerson()
```

Comment:
this will work as it is supposed when compensating writes project is finished, when this is finished on server we will have to update our baas version and test it against it